xbmc/filesystem/CurlFile.cpp

```diff
@@ -540,7 +540,7 @@ void CCurlFile::SetCommonOptions(CReadState* state)
   if (m_postdataset)
   {
     g_curlInterface.easy_setopt(h, CURLOPT_POST, 1 );
-    g_curlInterface.easy_setopt(h, CURLOPT_POSTFIELDSIZE, m_postdata.length());
+    g_curlInterface.easy_setopt(h, CURLOPT_POSTFIELDSIZE, m_postdata.size());
```

Comment:
Something else: https://curl.haxx.se/libcurl/c/CURLOPT_POSTFIELDSIZE.html we need the size in bytes anyways in the described usecase.